src/ui/EditorNS/editor.cpp

```diff
@@ -206,10 +236,27 @@ namespace EditorNS
 
     QList<QMap<QString, QString>> Editor::languages()
     {
+        // TODO: Cache this data.
+        QJSEngine engine;
+        QFileInfo fileInfo(Notepadqq::editorPath());
+        QString fileName = fileInfo.absolutePath() + "/classes/Languages.js";
```

Comment:
If we could find a way to parse the Languages.js file on build and create a readable config file for languages... this would be the optimal method... It would probably be faster than full JS parsing as well.  Alternatively, we could maintain the file manually, but that adds developer overhead.